### PR TITLE
refactor(website): cleanup pages file structure

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Move files around to have a cleaner file architecture.

Before

```
packages/website/src
├── plasma
│   ├── Home.tsx
│   ├── NotFound.tsx
│   ├── routes
│   │   ├── display-and-utilities
│   │   │   ├── index.tsx
│   │   │   └── pages
│   │   │       ├── BordersExamples.tsx
│   │   │       └── CollapsibleExamples.tsx
│   │   ├── feedback-and-info
│   │   │   ├── index.tsx
│   │   │   └── pages
│   │   │       ├── BadgeExamples.tsx
│   │   │       ├── FeedbackExamples.tsx
│   │   │       ├── InfoBoxExamples.tsx
│   │   │       ├── InfoTokenExamples.tsx
│   │   │       ├── LimitExamples.tsx
│   │   │       ├── LoadingExamples.tsx
│   │   │       ├── PopoverExamples.tsx
│   │   │       ├── ProgressBarExamples.tsx
│   │   │       ├── StatusCardExamples.tsx
│   │   │       ├── StepProgressBarExamples.tsx
│   │   │       ├── SyncFeedbackExamples.tsx
│   │   │       ├── ToastConnectedExamples.tsx
│   │   │       ├── ToastContentExamples.tsx
│   │   │       ├── ToastExamples.tsx
│   │   │       └── TooltipExamples.tsx
``` 

After
```
packages/website/src
├── pages
│   ├── Home.tsx
│   ├── NotFound.tsx
│   ├── display-and-utilities
│   │   ├── BordersExamples.tsx
│   │   ├── CollapsibleExamples.tsx
│   │   └── index.tsx
│   ├── feedback-and-info
│   │   ├── BadgeExamples.tsx
│   │   ├── FeedbackExamples.tsx
│   │   ├── InfoBoxExamples.tsx
│   │   ├── InfoTokenExamples.tsx
│   │   ├── LimitExamples.tsx
│   │   ├── LoadingExamples.tsx
│   │   ├── PopoverExamples.tsx
│   │   ├── ProgressBarExamples.tsx
│   │   ├── StatusCardExamples.tsx
│   │   ├── StepProgressBarExamples.tsx
│   │   ├── SyncFeedbackExamples.tsx
│   │   ├── ToastConnectedExamples.tsx
│   │   ├── ToastContentExamples.tsx
│   │   ├── ToastExamples.tsx
│   │   ├── TooltipExamples.tsx
│   │   └── index.tsx
```

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/website/src/pages/display-and-utilities/BordersExamples.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
 
-import VaporComponent from '../../../../demo-building-blocs/VaporComponent';
+import VaporComponent from '../../building-blocs/VaporComponent';
```
Comment (aiheon) [inline packages/website/src/pages/display-and-utilities/BordersExamples.tsx:3]: Next name to change :see_no_evil:

**Review by aiheon — APPROVED** [review]

File: packages/website/webpack.config.js [hunk 2]
```diff
@@ -24,8 +24,8 @@ module.exports = {
         extensions: ['.js', '.jsx', 'wasm'],
         alias: {
             '@coveord/plasma-react$': path.resolve(__dirname, '../react/dist/cjs/Entry.js'),
-            '@demo-styling': path.resolve(__dirname, 'src/demo-styling'),
-            '@routes': path.resolve(__dirname, 'src/plasma/routes'),
+            '@styles': path.resolve(__dirname, 'src/styles'),
+            '@pages': path.resolve(__dirname, 'src/pages'),
```
Comment (toofff) [inline packages/website/webpack.config.js:28]: the `@` is the new convention for JS dependencies but for folders that belong to the project, wouldn't it be better not to use it?

File: packages/website/src/utils/TableHOCExampleUtils.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
+import {DateUtils, TableRowConnected, TableRowNumberColumn} from '@coveord/plasma-react';
 import moment from 'moment';
 import * as React from 'react';
-import {DateUtils, TableRowConnected, TableRowNumberColumn} from '@coveord/plasma-react';
 
-import {IExampleRowData} from '../routes/layout/pages/TableHOCExamples';
-import {TableHOCServerExampleId} from '../routes/layout/pages/TableHOCServerExamples';
+import {IExampleRowData} from '../pages/layout/TableHOCExamples';
+import {TableHOCServerExampleId} from '../pages/layout/TableHOCServerExamples';
```
Comment (toofff) [inline packages/website/src/utils/TableHOCExampleUtils.tsx:6]: Even if I prefer not to prefix the folders with an @ when they belong to the project

```suggestion
import {IExampleRowData} from '@pages/layout/TableHOCExamples';
import {TableHOCServerExampleId} from '@pages/layout/TableHOCServerExamples';
```

File: packages/website/webpack.config.js (hunk 2 above)
Comment (toofff) [reply packages/website/webpack.config.js:28]: You can do the same for all the folders which is at the root of the `src` folder

**Review by toofff — APPROVED** [review]

File: packages/website/webpack.config.js (hunk 2 above)
Comment (gdostie) [reply packages/website/webpack.config.js:28]: This alias is needed just for this specific line https://github.com/coveo/plasma/blob/2c0b9e70049f39cae4ce8ef72bce522ebd24a536/packages/website/src/building-blocs/useCodeExample.tsx#L25

File: packages/website/src/utils/TableHOCExampleUtils.tsx (hunk 3 above)
Comment (gdostie) [reply packages/website/src/utils/TableHOCExampleUtils.tsx:6]: I don't want to use the alias everywhere, its only used for loading tsx files as raw files

**Pull request merged**